runtime/vam/materialize.go

```diff
@@ -25,19 +26,22 @@ func (m *Materializer) Pull(done bool) (zbuf.Batch, error) {
 	if vec == nil || err != nil {
 		return nil, err
 	}
-	b := vec.NewBuilder()
-	typ := vec.Type()
+	variant, _ := vec.(*vector.Variant)
+	var typ zed.Type
+	if variant == nil {
+		typ = vec.Type()
+	}
 	builder := zcode.NewBuilder()
 	var vals []zed.Value
-	for {
-		if !b(builder) {
-			return zbuf.NewArray(vals), nil
+	n := vec.Len()
+	for slot := uint32(0); slot < n; slot++ {
+		vec.Serialize(builder, slot)
```

Comment:
This more just a curiosity but going forward with @nwt's ongoing work on native Values do we imagine that some point soon we won't want to call Serialize here and instead build using native values?

Reply:
Yes, will need a better name for the method then.